### PR TITLE
[CAMEL-15094] Create an Endpoint DSL archetype

### PR DESCRIPTION
**Pull request by PickledDragon**
- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

---
The MR adds a new archetype for Endpoint DSL as requested in [CAMEL-15094](https://issues.apache.org/jira/browse/CAMEL-15094).
Fairly similar to existing archetypes like `camel-archetype-main` but illustrates the usage of strongly typed [Endpoint DSL](https://camel.apache.org/manual/latest/Endpoint-dsl.html)

### CODE REVIEW

**Review by oscerd — APPROVED** [review]
Thanks

File: archetypes/camel-archetype-endpointdsl/src/main/resources/archetype-resources/src/main/java/MyRouteBuilder.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.builder.endpoint.dsl.FileEndpointBuilderFactory;
+
+import static org.apache.camel.builder.endpoint.StaticEndpointBuilders.file;
+import static org.apache.camel.builder.endpoint.StaticEndpointBuilders.timer;
+/**
+ * A Camel Java DSL Router
+ */
+public class MyRouteBuilder extends RouteBuilder {
```
Comment (davsclaus) [inline archetypes/camel-archetype-endpointdsl/src/main/resources/archetype-resources/src/main/java/MyRouteBuilder.java:27]: Its better to extend EndpointRouteBuilder, then you dont need those static imports, see for example https://github.com/apache/camel/blob/4ea9e6c357371682b855d2d79655b41120331b7a/core/camel-endpointdsl/src/test/java/org/apache/camel/builder/endpoint/FileDelayTest.java#L50

File: archetypes/camel-archetype-endpointdsl/src/main/resources/archetype-resources/src/main/java/MyRouteBuilder.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+## limitations under the License.
+## ------------------------------------------------------------------------
+package ${package};
+
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.builder.endpoint.dsl.FileEndpointBuilderFactory;
+
+import static org.apache.camel.builder.endpoint.StaticEndpointBuilders.file;
+import static org.apache.camel.builder.endpoint.StaticEndpointBuilders.timer;
+/**
```
Comment (davsclaus) [inline archetypes/camel-archetype-endpointdsl/src/main/resources/archetype-resources/src/main/java/MyRouteBuilder.java:24]: Add empty line above

**Review by davsclaus — CHANGES_REQUESTED** [review]

File: archetypes/camel-archetype-endpointdsl/src/main/resources/archetype-resources/src/main/java/MyRouteBuilder.java (hunk 1 above)
Comment (PickledDragon) [reply archetypes/camel-archetype-endpointdsl/src/main/resources/archetype-resources/src/main/java/MyRouteBuilder.java]: Commit [77cd11e](https://github.com/apache/camel/pull/4542/commits/77cd11e9e2954d9723499329159ab5b2ff042825) resolves both observations. Thank you for your support.

**Review by davsclaus — APPROVED** [review]

**Pull request merged**